src/main/webapp/components/navbar.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+  <nav class="blue nav-bar">
+    <a href="index.html" class="bold" id="title">starfish</a>
+    <img src="../assets/starfish.png" id="logo">
+    <div id="action-items">
+      <a href="#" id="post-action">PostANote</a>
+      <div class="dropdown">
+        <button class="dropbtn" @click.stop="toggleDropdown">
+          <img id="user-profile" src="../assets/user.svg">
+        </button>
+        <div class="dropdown-content" v-if="showDropdown" v-click-outside="hideDropdown">
+          <img src="../assets/user.svg" height="115px"/>
+          <p class="bold" id="dropdown-name">{{ user.name }}</p>
+          <p id="dropdown-points">{{ user.points }} points</p>
+          <a class="dropdown-link" href="#">My Profile</a> 
+          <a class="dropdown-link" href="#">Favorite Notes</a>
+          <a class="dropdown-link" href="#">Logout</a>
+        </div>
+      </div>
+    </div> 
+  </nav>
+</template>
+
+<script>
+Vue.directive('click-outside', {
+  bind: function (el, binding, vnode) {
+    this.event = function(event) {
+      if(!(el == event.target || el.contains(event.target))) {
+        vnode.context[binding.expression](event);
+      }
+    };
+    document.body.addEventListener('click', this.event);
+  },
+  unbind: function(el) {
+    document.body.removeEventListener('click', this.event);
+  }
+});
+
+module.exports = {
+  data: function() {
+    return {
+      showDropdown: false
+    }
+  },
+  props: {
+    user: Object
+  },
+  methods: {
+    hideDropdown() {
+      this.showDropdown = false;
+    },
+    toggleDropdown() {
+      this.showDropdown = !this.showDropdown;
+    }
+  }
+}
+</script>
+
+<style>
+.bold {
+  font-weight: bold;
+}
+
+.blue {
+  background-color: #004aad;
+}
+
```

Comment:
I'm wondering if it might be a good idea to also add a "sticky" class and adding that to the navbar as well to make sure the navbar position remains fixed no matter where the user scrolls.
```suggestion
.sticky { position: fixed; }
```

Reply:
I'll add that!